packages/flutter/lib/src/widgets/shared_state.dart

```diff
@@ -0,0 +1,197 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'framework.dart';
+
+/// A widget to manage shared state for its descendants.
+///
+/// This can be used for managing state when multiple widgets need their state
+/// to be kept in sync. It is most useful when the closest common widget to the
+/// widgets that need to share the state is several levels above the widgets
+/// that are using it. When the state is close to the widgets using them, it is
+/// often preferable to simply pass the needed state to the descendants that
+/// need it instead of using this widget.
+///
+/// For complex applications, it may be more appropriate to use a package that
+/// provides more features and a stream-oriented paradigm like Rx or BLoC.
```

Comment:
BLoC isn't a searchable / known concept. I wouldn't reference it.

Reply:
OK.  Removed.